### PR TITLE
[FE](styles): real legend UI [GMW-727]

### PR DESCRIPTION
**Pull request by anamontiaga**
### Designs

[Figma](https://www.figma.com/proto/GJCx8yGfx6JERPuxlpG3nK/Mangrove-Watch-%5BInternal%5D?node-id=4343-26935&scaling=min-zoom&page-id=2772%3A20372&starting-point-node-id=2772%3A21073&show-proto-sidebar=1)

### Feature relevant tickets

[GMW-727](https://vizzuality.atlassian.net/browse/GMW-727)

[GMW-727]: https://vizzuality.atlassian.net/browse/GMW-727?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +54,7 @@
     "@turf/bbox": "6.5.0",
     "@types/d3-format": "^3.0.1",
     "@types/lodash-es": "4.17.7",
+    "add": "^2.0.6",
```
Comment (mluena) [inline package.json:57]: remove this dependency

File: package.json [hunk 2]
```diff
@@ -78,6 +80,7 @@
     "recoil-sync": "0.2.0",
     "shadcn-ui": "latest",
     "sharp": "^0.32.6",
+    "slider": "^1.0.4",
```
Comment (mluena) [inline package.json:83]: do we need both?

File: src/components/collapsible/index.tsx [hunk 3]
```diff
@@ -3,9 +3,9 @@ import { useCallback, useState } from 'react';
 import * as Collapsible from '@radix-ui/react-collapsible';
 import { ChevronUpIcon } from '@radix-ui/react-icons';
 import cn from 'classnames';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/components/collapsible/index.tsx:6]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/components/slider/index.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import cn from 'lib/classnames';
+
+import * as SliderPrimitive from '@radix-ui/react-slider';
+
+const Slider = ({ defaultValue, max = 1, step = 0.1, onValueChange, className }) => {
+  const [currentValue, setCurrentValue] = useState(defaultValue);
+
+  return (
+    <form>
```
Comment (mluena) [inline src/components/slider/index.tsx:11]: Why the form?

File: src/components/slider/index.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+
+import cn from 'lib/classnames';
+
+import * as SliderPrimitive from '@radix-ui/react-slider';
+
+const Slider = ({ defaultValue, max = 1, step = 0.1, onValueChange, className }) => {
```
Comment (mluena) [inline src/components/slider/index.tsx:7]: Not sure about this.

const Slider = forwardRef<
  ElementRef<typeof SliderPrimitive.Root>,
  ComponentPropsWithoutRef<typeof SliderPrimitive.Root>
>(({ className, ...props }, ref) => (
  <SliderPrimitive.Root
    ref={ref}
    className={cn('relative flex h-4 w-full touch-none select-none items-center': true,
          [className]: !!className)}
    {...props}
  >

File: src/containers/datasets/alerts/hooks.tsx [hunk 6]
```diff
@@ -12,6 +12,7 @@ import { analysisAtom } from 'store/analysis';
 import { alertsEndDate, alertsStartDate } from 'store/widgets/alerts';
 
 import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/alerts/hooks.tsx:15]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/biomass/hooks.tsx [hunk 7]
```diff
@@ -12,6 +12,7 @@ import { BiomassYearSettings } from 'store/widgets/biomass';
 
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
 import { AxiosError, CanceledError } from 'axios';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/biomass/hooks.tsx:15]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/biomass/map-legend.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+      {config?.legend?.items?.map(({ color, label }) => {
+        return (
+          <div key={label} className="flex items-start">
+            <div
+              style={{ backgroundColor: color }}
+              className="my-0.5 mr-2.5 h-4 w-2 shrink-0 rounded-md text-sm"
+            />
+            <div className="flex flex-col items-start text-sm">
+              <p>
+                {label} <span className="text-xs">tons/ha</span>
```
Comment (mluena) [inline src/containers/datasets/biomass/map-legend.tsx:17]: unit could be dynamic as well

File: src/containers/datasets/blue-carbon/hooks.tsx [hunk 9]
```diff
@@ -13,6 +13,7 @@ import { drawingToolAtom } from 'store/drawing-tool';
 
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
 import { AxiosError, CanceledError } from 'axios';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/blue-carbon/hooks.tsx:16]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/contextual-layers/global-tidal-wetland-change/hooks.tsx [hunk 10]
```diff
@@ -1,5 +1,7 @@
 import type { SourceProps, LayerProps } from 'react-map-gl';
 
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/contextual-layers/global-tidal-wetland-change/hooks.tsx:3]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/contextual-layers/salt-marsh/hooks.tsx [hunk 11]
```diff
@@ -1,5 +1,7 @@
 import type { SourceProps, LayerProps } from 'react-map-gl';
 
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/contextual-layers/salt-marsh/hooks.tsx:3]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/contextual-layers/tidal-flats/hooks.tsx [hunk 12]
```diff
@@ -1,5 +1,7 @@
 import type { SourceProps, LayerProps } from 'react-map-gl';
 
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/contextual-layers/tidal-flats/hooks.tsx:3]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/drivers-change/hooks.tsx [hunk 13]
```diff
@@ -6,6 +6,7 @@ import { useRouter } from 'next/router';
 
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
 import { AxiosResponse } from 'axios';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/drivers-change/hooks.tsx:9]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/fisheries/hooks.tsx [hunk 14]
```diff
@@ -5,6 +5,7 @@ import { useRouter } from 'next/router';
 import { numberFormat, formatAxis } from 'lib/format';
 
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/fisheries/hooks.tsx:8]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/habitat-extent/hooks.tsx [hunk 15]
```diff
@@ -11,6 +11,7 @@ import { drawingToolAtom } from 'store/drawing-tool';
 
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
 import { AxiosError, CanceledError } from 'axios';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/habitat-extent/hooks.tsx:14]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/height/hooks.tsx [hunk 16]
```diff
@@ -12,6 +12,7 @@ import { drawingToolAtom } from 'store/drawing-tool';
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
 import { AxiosError, CanceledError } from 'axios';
 import { AxiosResponse } from 'axios';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/height/hooks.tsx:15]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/national-dashboard/indicator-source/index.tsx [hunk 17]
```diff
@@ -81,7 +81,7 @@ const IndicatorSource = ({
           year: yearSelected,
         },
       });
-      const widgetsUpdate = new Set(widgetsCheck);
+      const widgetsUpdate = new Set(widgetsCheck) as any;
```
Comment (mluena) [inline src/containers/datasets/national-dashboard/indicator-source/index.tsx:84]: avoid any, have you tried with Partial?

File: src/containers/datasets/national-dashboard/widget.tsx [hunk 18]
```diff
@@ -45,6 +45,7 @@ const NationalDashboard = () => {
                 {/* <h3 className={WIDGET_SUBTITLE_STYLE}>{indicator}</h3> */}
                 {sources.map(({ source, years, unit, data_source }) => {
                   const dataSource = data_source.find((d) => d.year === currentYear);
+                  console.log({ dataSource });
```
Comment (mluena) [inline src/containers/datasets/national-dashboard/widget.tsx:48]: remove

File: src/containers/datasets/net-change/hooks.tsx [hunk 19]
```diff
@@ -14,6 +14,7 @@ import { useQuery, UseQueryOptions } from '@tanstack/react-query';
 import { AxiosError, CanceledError } from 'axios';
 import { AxiosResponse } from 'axios';
 import { format } from 'd3-format';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/net-change/hooks.tsx:17]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/protection/hooks.tsx [hunk 20]
```diff
@@ -7,6 +7,7 @@ import { useRouter } from 'next/router';
 import { numberFormat } from 'lib/format';
 
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/protection/hooks.tsx:10]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/restoration/hooks.tsx [hunk 21]
```diff
@@ -2,14 +2,24 @@ import { useMemo } from 'react';
 
 import type { SourceProps, LayerProps } from 'react-map-gl';
 
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/restoration/hooks.tsx:5]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/restoration/map-legend.tsx [hunk 22]
```diff
@@ -0,0 +1,20 @@
+const RestorationMapLegend = () => {
+  return (
+    <div className="w-[260px]">
+      <div className="flex  justify-between text-black/85">
+        <p>0</p>
+        <p>100</p>
```
Comment (mluena) [inline src/containers/datasets/restoration/map-legend.tsx:6]: why are these numbers static?

File: src/containers/datasets/species-distribution/hooks.tsx [hunk 23]
```diff
@@ -6,6 +6,7 @@ import { useRouter } from 'next/router';
 
 import { useQuery, UseQueryOptions } from '@tanstack/react-query';
 import { AxiosResponse } from 'axios';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/species-distribution/hooks.tsx:9]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/species-location/hooks.tsx [hunk 24]
```diff
@@ -8,6 +8,7 @@ import { SpeciesLocationState } from 'store/widgets/species-location';
 
 import { useQuery, QueryClient } from '@tanstack/react-query';
 import type { QueryObserverOptions } from '@tanstack/react-query';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/species-location/hooks.tsx:11]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/map/legend/index.tsx [hunk 25]
```diff
@@ -1,106 +1,312 @@
...
-                  </button>
-                </div>
-              )}
-            </Helper>
-          );
-        })}
-    </div>
+                      <div className="flex w-full items-center justify-between">
+                        <div className="flex space-x-2">
+                          <button type="button" onClick={() => console.log('order layer')}>
```
Comment (mluena) [inline src/containers/map/legend/index.tsx:174]: ???

File: src/lib/utils/index.tsx [hunk 26]
```diff
@@ -1,8 +1,10 @@
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/lib/utils/index.tsx:1]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/store/layers/index.ts [hunk 27]
```diff
@@ -1,4 +1,5 @@
-import { array, object, string, number } from '@recoiljs/refine';
+import { array, object, string, stringLiterals } from '@recoiljs/refine';
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/store/layers/index.ts:2]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

**Review by mluena — CHANGES_REQUESTED** [review]

**Comment by mluena** [comment]
Info button should show the widget in a Dialog, to highlight that part, not the info. Maybe we could change that after finishing and merging this PR

File: src/containers/datasets/restoration/map-legend.tsx (hunk 22 above)
Comment (anamontiaga) [reply src/containers/datasets/restoration/map-legend.tsx:6]: Are percentages

File: src/components/slider/index.tsx (hunk 5 above)
Comment (anamontiaga) [reply src/components/slider/index.tsx:7]: Think this is inside radix ui component

Comment (mluena) [reply src/components/slider/index.tsx:7]: use shadcn https://ui.shadcn.com/docs/components/slider

File: src/components/widget-controls/index.tsx [hunk 28]
```diff
@@ -16,8 +16,8 @@ import { useWidgets } from 'containers/widgets/hooks';
 
 import { SwitchWrapper, SwitchRoot, SwitchThumb } from 'components/switch';
 import { breakpoints } from 'styles/styles.config';
+import { ActiveLayers } from 'types/layers';
```
Comment (mluena) [inline src/components/widget-controls/index.tsx:19]:
```suggestion
import type { ActiveLayers } from 'types/layers';
```

File: src/containers/datasets/contextual-layers/allen-coral-reef/hooks.tsx [hunk 29]
```diff
@@ -1,5 +1,7 @@
 import type { SourceProps, LayerProps } from 'react-map-gl';
 
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/containers/datasets/contextual-layers/allen-coral-reef/hooks.tsx:3]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/contextual-layers/global-tidal-wetland-change/hooks.tsx (hunk 10 above)
Comment (mluena) [inline src/containers/datasets/contextual-layers/global-tidal-wetland-change/hooks.tsx:3]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/containers/datasets/flood-protection/widget.tsx [hunk 30]
```diff
@@ -23,8 +22,8 @@ import {
   WIDGET_SUBTITLE_STYLE,
   WIDGET_SELECT_STYLES,
 } from 'styles/widgets';
+import { ActiveLayers } from 'types/layers';
```
Comment (mluena) [inline src/containers/datasets/flood-protection/widget.tsx:25]:
```suggestion
import type { ActiveLayers } from 'types/layers';
```

File: src/containers/datasets/national-dashboard/indicator-source/index.tsx [hunk 31]
```diff
@@ -12,6 +12,7 @@ import Icon from 'components/icon';
 import { Popover, PopoverContent, PopoverTrigger } from 'components/popover';
 import { SwitchWrapper, SwitchRoot, SwitchThumb } from 'components/switch';
 import WidgetControls from 'components/widget-controls';
+import { ActiveLayers } from 'types/layers';
```
Comment (mluena) [inline src/containers/datasets/national-dashboard/indicator-source/index.tsx:15]:
```suggestion
import type { ActiveLayers } from 'types/layers';
```

File: src/containers/datasets/national-dashboard/indicator-source/index.tsx [hunk 32]
```diff
@@ -81,7 +82,7 @@ const IndicatorSource = ({
           year: yearSelected,
         },
       });
-      const widgetsUpdate = new Set(widgetsCheck);
+      const widgetsUpdate = new Set(widgetsCheck) as unknown as ActiveLayers[];
```
Comment (mluena) [inline src/containers/datasets/national-dashboard/indicator-source/index.tsx:85]:
```suggestion
"TO - DO -review types"
      const widgetsUpdate = new Set(widgetsCheck) as unknown as ActiveLayers[];
```

File: src/types/layers.ts [hunk 33]
```diff
@@ -1,9 +1,19 @@
 import type { LayerProps as RMGLLayerProps } from 'react-map-gl';
 
+import { Visibility } from 'mapbox-gl';
```
Comment (mluena) [inline src/types/layers.ts:3]:
```suggestion
import type { Visibility } from 'mapbox-gl';
```

File: src/types/layers.ts [hunk 34]
```diff
@@ -1,9 +1,19 @@
...
+
+import { ContextualBasemapsId, WidgetSlugType } from 'types/widget';
```
Comment (mluena) [inline src/types/layers.ts:5]:
```suggestion
import type { ContextualBasemapsId, WidgetSlugType } from 'types/widget';
```

**Review by mluena — APPROVED** [review]

**Pull request merged**